app/src/main/java/net/skyscanner/backpack/demo/compose/FloatingNotificationStory.kt

```diff
@@ -0,0 +1,75 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.demo.compose
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import kotlinx.coroutines.launch
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.floatingnotification.BpkFloatingNotification
+import net.skyscanner.backpack.compose.floatingnotification.Cta
+import net.skyscanner.backpack.compose.floatingnotification.rememberBpkFloatingNotificationState
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.Heart
+import net.skyscanner.backpack.demo.R
+
+@Composable
+fun FloatingNotificationStory() {
+  val scope = rememberCoroutineScope()
+  val textOnlyState = rememberBpkFloatingNotificationState()
+  val longTextWithIconState = rememberBpkFloatingNotificationState()
+  val textWithCtaState = rememberBpkFloatingNotificationState()
+  val allState = rememberBpkFloatingNotificationState()
+
+  BpkButton(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(BpkSpacing.Base),
+    text = stringResource(id = R.string.floating_notification_show),
+    enabled = !textOnlyState.visible && !longTextWithIconState.visible && !textWithCtaState.visible && !allState.visible
+  ) {
+    scope.launch {
+      textOnlyState.show(
+        text = "Lorem ipsum dolor sit amet."
```

Comment:
can you use string resources for these?

Reply:
by using `stringResource()` it gives an error `@Composable invocations can only happen from the context of a @Composable function` because button's `onClick()` does not receive a @Composable. Is there a way to use string resources here?

Reply:
I believe you have to define a val outside of this scope and use it within

Reply:
thanks. changed them